### ISSUE TITLE
infinite loop with fortran parser on C file

### ISSUE THREAD

**Issue by kcwu** [universal-ctags/ctags#629]
How to reproduce
```
./ctags.20151012 --language-force=fortran ctags/Units/parser-c.r/enum.c.d/input.c
```

It's found by forcing language as fortran against universal-ctags unit test files.

**Comment by kcwu** [comment]
and ```ctags/Units/cxx11enum.cpp.d/input.cpp```

**Comment by masatake** [comment]
BTW You can do the same with fuzz target:
```
b% make fuzz LANGUAGES=Fortran
make fuzz LANGUAGES=Fortran
Semi-fuzzing (Fortran)                                      
............................................
[timeout Fortran]                       ./Units/cxx11enum.cpp.d/input.cpp

Makefile:399: recipe for target 'fuzz' failed
make: *** [fuzz] Error 1
```

(I'm looking for better name for the target. It is not "fuzz".)

**Comment by masatake** [comment]
SRHINK=1 spots the token.

```
b% make fuzz LANGUAGES=Fortran SHRINK=1
make fuzz LANGUAGES=Fortran SHRINK=1
Semi-fuzzing (Fortran)                                      
............................................
[timeout Fortran]                       ./Units/cxx11enum.cpp.d/input.cpp
Shrinking ./Units/cxx11enum.cpp.d/input.cpp as Fortran
............................................................
[0        351]...timeout(124)
[0          0]...ok(0)
[0        351]...timeout(124)
[0        176]...timeout(124)
[0         89]...timeout(124)
[0         46]...timeout(124)
[0         25]...timeout(124)
[0         15]...timeout(124)
[0         10]...timeout(124)
[0          8]...timeout(124)
[0          7]...timeout(124)
[0          6]...timeout(124)
[0          5]...timeout(124)
[0          4]...ok(0)
[0          5]...timeout(124)
[2          5]...ok(0)
[1          5]...timeout(124)
Minimal badinput: ./Units/cxx11enum.cpp.d/SHRINK-Fortran.tmp
............................................................
enum

Makefile:399: recipe for target 'fuzz' failed
make: *** [fuzz] Error 1
```

**Comment by QingmingHe** [comment]
fixed in #630
